### DISCUSSION TITLE
[RFC] Thoughts on `on_init_start` and `on_init_end` hooks

### DISCUSSION THREAD

**Discussion by daniellepintz** [Lightning-AI/pytorch-lightning#10677] [Lightning Trainer API: Trainer, LightningModule, LightningDataModule] [answered]
These hooks are called when trainer initialization begins and ends, before the model has been set, essentially allowing the user to modify the Trainer constructor.  Should we be giving the user this much control over Trainer constructor? Are there scenarios where this is needed? Or can we deprecate these hooks?

https://github.com/PyTorchLightning/pytorch-lightning/blob/338f3cf63686935355c749920b2f298f3d18a26f/pytorch_lightning/trainer/callback_hook.py#L55-L63

cc @ananthsub

**Comment by ananthsub** [comment, marked as answer]
@carmocca @tchaton @awaelchli do you know how these hooks are used? Have you seen any examples of these being used by the community? These hooks go way way back, but I can't think of when they'd be needed given the user "owns" the Trainer initialization. It's also unclear when `on_init_start` actually happens: does that mean callbacks should be the first thing initialized? 

it seems a lot more straightforward to write this:
```python
trainer = Trainer(...)
run_all_my_fancy_logic_now(trainer)

# use the trainer here
```

Let's discuss in https://github.com/PyTorchLightning/pytorch-lightning/issues/10894

**Reply by awaelchli** [reply]
have replied in the linked issue. They were there since a long time. Currently used by three of our own callbacks.
